### PR TITLE
create_test: add --driver support for e3sm

### PR DESCRIPTION
**Pull request by jgfouca**
Also, fix a couple issues in driver handling in test_scheduler.

Test suite:
Test baseline:
Test namelist changes:
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?:

Update gh-pages html (Y/N)?:

### CODE REVIEW

File: CIME/test_scheduler.py [hunk 1]
```diff
@@ -681,15 +683,16 @@ def _create_newcase_phase(self, test):
                         )
                     )
                 elif case_opt.startswith("V"):
-                    self._cime_driver = case_opt[1:]
```
Comment (jgfouca) [inline CIME/test_scheduler.py:684]: The original code looks like a bug. The presence of one test with Vdriver would change the driver for all the tests.

File: CIME/test_scheduler.py [hunk 2]
```diff
@@ -924,7 +932,9 @@ def _xml_phase(self, test):
                 elif opt.startswith("A"):
                     # A option is for testing in ASYNC IO mode, only available with nuopc driver and pio2
                     envtest.set_test_parameter("PIO_ASYNC_INTERFACE", "TRUE")
-                    envtest.set_test_parameter("CIME_DRIVER", "nuopc")
```
Comment (jgfouca) [inline CIME/test_scheduler.py:927]: This also looked really problematic. Setting the driver to nuopc even if that doesn't match cime_driver?

**Comment by jgfouca** [comment]
@jedwards4b , is this error `ERROR: ESMF version should be 8.6.1 or newer` expected in CESM?

File: CIME/test_scheduler.py (hunk 1 above)
Comment (jedwards4b) [reply CIME/test_scheduler.py:684]: That's been there for a long time, seems like we would have seen problems by now if that were the case.   On the other hand, I agree.

File: CIME/test_scheduler.py (hunk 2 above)
Comment (jedwards4b) [reply CIME/test_scheduler.py:927]: PIO async mode is only supported in nuopc - but since cesm only supports the nuopc driver now, this is okay to remove.

**Review by jedwards4b — COMMENTED** [review]
No, I don't think that this is expected - this was passing in the previous several PR's - 
` ESMF version should be 8.6.1 or newer expected in CESM?`. Did something in docker regress somehow?

**Comment by jgfouca** [comment]
@jedwards4b , would you mind trying a CESM test on this branch? I don't see how this PR is causing an ESMF problem.

**Comment by jedwards4b** [comment]
According to the docker README: `docker/README.md:| ESMF_VERSION | Version of ESMF, the default will install the latest | 8.2.0`
I don't see it set anywhere so you think that the latest would be used - the latest is 8.6.1.
https://anaconda.org/conda-forge/esmf says v8.6.1

**Comment by jedwards4b** [comment]
@jgfouca I don't either and I think that we may need to rethink this docker testing framework - it seems like it's been a problem in nearly every cime PR since it was introduced.  Testing on derecho now...

**Comment by jedwards4b** [comment]
cesm tests are passing locally -

**Review by jedwards4b — APPROVED** [review]
Ran cesm tests on derecho with no issues - the .github fail seems to be a docker issue.

File: CIME/test_scheduler.py (hunk 1 above)
Comment (rljacob) [reply CIME/test_scheduler.py:684]: You mean adding one Vdriver a test in a suite would change all tests in that suite?

Comment (jgfouca) [reply CIME/test_scheduler.py:684]: @rljacob , yes. `self._cime_driver` is shared for all tests.

**Comment by jgfouca** [comment]
I see the same failure on a PR that changed nothing: https://github.com/ESMCI/cime/pull/4674

Merging.

**Pull request merged**